tests/testthat/test-unnecessary_nesting_linter.R

```diff
@@ -345,3 +345,255 @@ test_that("lints vectorize", {
     unnecessary_nesting_linter()
   )
 })
+
+test_that("unnecessary_nesting_linter skips allowed usages", {
+  linter <- unnecessary_nesting_linter()
+
+  expect_lint(
+    trim_some("
```

Comment:
These usages are inconsistent WRT indentation and content of the last line across the file.
Do you mind cleaning up so we have
```r
trim_some("
  two-spaces-indented
  code
  here
") # but not here
```
everywhere in the file?